README.md

```diff
@@ -10,33 +10,58 @@ Telepoison is a [opentelemetry-instrumented](https://github.com/open-telemetry/o
 
 ## Usage
 
-Simply replace `HTTPoison` with `Telepoison` when calling one of the request methods (get(), get!(), post(), request(), etc.)
+Call `Telepoison.setup/1` within your application start-up.
+
+Replace usages of the `HTTPoison` module with `Telepoison` when calling one of the *derived* request functions provided by `HTTPoison` (`HTTPoison.get/3`, `HTTPoison.get!/3` etc.)
 
 ```elixir
-# before
+# Before
 HTTPoison.get!(url, headers, opts)
 
-# after
+# After
 Telepoison.get!(url, headers, opts)
 ```
 
-Additionally, telepoison adds some options that can be passed in the `opts` HTTPoison argument to set OpenTelemetry-related stuff.
-These start with `:ot_`
+## Configuration
+
+`Telepoison.setup/1` takes a `Keyword list` that can configure if and how the `http.route` Open Telemetry metadata will be set per request using the `:infer_route` option.
+
+* If no value is provided then the out of the box, conservative inference provided by `Telepoison.URI.infer_route_from_request/1` is used to determine the inference
+
+* If an function with an arity of 1 (the `t:HTTPoison.Request/0` `request`) is provided then that function is used to determine the inference
```

Comment:
```suggestion
* If a function with an arity of 1 (the `t:HTTPoison.Request/0` `request`) is provided then that function is used to determine the inference
```

Reply:
Also, if feel it could be clearer that `HTTPoison.Request` is the input of the unary function

Reply:
Updated, let me know if that's clearer!